File: Stacks/RemoveConsecEleFromString/removeConsec.py
class Solution:
    def removeDuplicates(self, s: str) -> str:
        n = len(s)
        if n==0: return ""
        st = []
        st.append(s[0])
        for i in range(1,n):
            if not len(st) or s[i]!=st[-1]:
                st.append(s[i])
            else:  
                st.pop()
        ans = ""
        for i in range(len(st)):
            ans += st[i]
        return ans

File: Stacks/RemoveConsecEleFromString/removeConsec.cpp
#include <string>
#include <stack>
using namespace std;

class Solution {
public:
    string removeDuplicates(string s) {
        int n = s.size();
        if(n==0)return "";
        stack<char>st;
        st.push(s[0]);
        for(int i=1;i<n;i++){
            if(!st.size() || st.top() != s[i]){
                st.push(s[i]);
            }
            else{
                st.pop();
            }
        }
        string ans = "";
        while(!st.empty()){
            char val = st.top();
            ans = string(1,val)+ans;
            st.pop();
        }
        return ans;
    }
};